src/box/lua/config/descriptions.lua

```diff
@@ -0,0 +1,2446 @@
+    since normally vinyl throttles inserts when it is running low on compaction
+    bandwidth. Compaction can also be initiated manually with
+    `index_object:compact()`.
+]])
+
+M['vinyl.write_threads'] = format_text([[
+    The maximum number of write threads that vinyl can use for some
+    concurrent operations, such as I/O and compression.
+]])
+
+-- }}} vinyl configuration
+
+-- {{{ wal configuration
+
+M['wal'] = format_text([[
+    This section defines configuration parameters related to write-ahead log.
+]])
+
+M['wal.cleanup_delay'] = format_text([[
+    The delay in seconds used to prevent the Tarantool garbage collector from
+    immediately removing write-ahead log files after a node restart. This
+    delay eliminates possible erroneous situations when the master deletes
+    WALs needed by replicas after restart. As a consequence, replicas sync
+    with the master faster after its restart and don't need to download all
+    the data again. Once all the nodes in the replica set are up and
+    running, a scheduled garbage collection is started again even if
+    `wal.cleanup_delay` has not expired.
+
+]])
+
+M['wal.dir'] = format_text([[
+    A directory where write-ahead log (`.xlog`) files are stored. A relative
+    path in this option is interpreted as relative to `process.work_dir`.
+
+    By default, WAL files and snapshots are stored in the same directory.
+    However, you can set different values for the `wal.dir` and `snapshot.dir`
+    options to store them on different physical disks for performance matters.
+]])
+
+M['wal.dir_rescan_delay'] = format_text([[
+    The time interval in seconds between periodic scans of the write-ahead-log
+    file directory, when checking for changes to write-ahead-log files for the
+    sake of replication or hot standby.
+]])
+
+M['wal.ext'] = format_text([[
+    This section describes options related to WAL extensions.
+]])
+
+M['wal.ext.new'] = format_text([[
+    Enable storing a new tuple for each CRUD operation performed. The option
+    is in effect for all spaces. To adjust the option for specific spaces,
+    use the `wal.ext.spaces` option.
+]])
+
+M['wal.ext.old'] = format_text([[
+    Enable storing an old tuple for each CRUD operation performed. The option
+    is in effect for all spaces. To adjust the option for specific spaces,
+    use the `wal.ext.spaces` option.
+]])
+
+M['wal.ext.spaces'] = format_text([[
+    Enable or disable storing an old and new tuple in the WAL record for a
+    given space explicitly. The configuration for specific spaces has
+    priority over the configuration in the `wal.ext.new` and `wal.ext.old`
+    options.
+
+    The option is a key-value pair:
+
+    - The key is a space name (string).
+    - The value is a table that includes two optional boolean options:
+    `old` and `new`. The format and the default value of these options are
+    described in `wal.ext.old` and `wal.ext.new`.
+]])
+
+M['wal.ext.spaces.*'] = 'Space name.'
```

Comment:
If we describe maping values (not keys), I would say something like the following:

> Per-space WAL extensions configuration.